Main.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections.Generic;
+
+// Author: Sonia Álvarez
+
+class Program
+{
+    static void Main()
+    {
+        // Probando que compila y que se usan tipos enumerados
+        int fuerzaAldeana = (int)TipoPersonaje.Aldeano;
+        var tipoAldeano = TipoPersonaje.Aldeano;
+
+        Console.WriteLine( $"La fuerza de cada unidad aldeana tiene valor: {fuerzaAldeana} ");
+        Console.WriteLine( $"Cada unidad aldeana es del tipo enumerado: {tipoAldeano} ");
+
+        // Toda Unidad tiene el mismo valor entero para su salud/vida
+        int vida = 20;
+
+        // Creando una lista vacía para el equipo Azul
+        List<Personaje> EquipoAzul = new List<Personaje>();
+
+        // Probando una instancia de la clase Personaje
+        Arquero pepe = new Arquero (
+            vida, 
+            (int)TipoPersonaje.Arquero;
+        );
+
+        // Agregando un Personaje a la lista del equipo Azul
+        EquipoAzul.Add( pepe );
+
+        EquipoAzul[0].Info();
+    }
+}
+
+
+// Valores constantes de las unidades de los equipos
+// Se asigna un entero que equivale a su valor de ataque
+
+enum TipoPersonaje {
```

Comment:
Un enumerado aquí solo trae problemas ya que no se está utilizando de la manera correcta y hay que hacer "ñapas" como los castings que estáis haciendo, que son un mal hábito de programación ya que son fuentes de bugs. Si queréis mantener una estructura similar deberíais utilizar un diccionario, aunque me parece un poco de sobreingeniería para la necesidad que tenemos aquí.